soroban-ledger-snapshot/src/lib.rs

```diff
@@ -118,6 +124,61 @@ impl LedgerSnapshot {
             }
         }
     }
+
+    pub fn add_contract_to_entries(&mut self, contract_id: [u8; 32], wasm_hash: [u8; 32]) {
+        let entries = &mut self.ledger_entries;
+        // Create the contract
+        let contract_key = LedgerKey::ContractData(LedgerKeyContractData {
+            contract_id: contract_id.into(),
+            key: ScVal::Static(ScStatic::LedgerKeyContractCode),
+        });
+
+        let contract_entry = LedgerEntry {
+            last_modified_ledger_seq: 0,
+            data: LedgerEntryData::ContractData(ContractDataEntry {
+                contract_id: contract_id.into(),
+                key: ScVal::Static(ScStatic::LedgerKeyContractCode),
+                val: ScVal::Object(Some(ScObject::ContractCode(ScContractCode::WasmRef(Hash(
+                    wasm_hash,
+                ))))),
+            }),
+            ext: LedgerEntryExt::V0,
+        };
+        for (k, e) in entries.iter_mut() {
+            if **k == contract_key {
+                **e = contract_entry;
+                return;
+            }
+        }
+        entries.push((Box::new(contract_key), Box::new(contract_entry)));
```

Comment:
Would it be reasonable to use the `update_entries` function above to do the write? It contains logic already for identifying an existing key and replacing it.